### ISSUE TITLE
Port release jenkins job to github actions

### ISSUE THREAD

**Issue by poshul** [OpenMS/OpenMS#6994]

**Comment by jpfeuffer** [comment]
Copy-paste scripts from
https://abibuilder.cs.uni-tuebingen.de/jenkins/job/openms/job/release/job/openms_release/

Maybe switch a bit more to "gh cli" commands instead of using direct requests to the API. But sometimes you want the response to get the link for a successful upload. Not sure if you can get that with "gh cli"

Add secrets to GitHub repository settings. https://docs.github.com/en/actions/security-guides/encrypted-secrets

Use them.

Add job parameters ("announce", "latest") as GitHub inputs. https://github.blog/changelog/2021-11-10-github-actions-input-types-for-manual-workflows/

For SSH steps in Jenkins you need additional secrets for SSH access to archive (creating symlinks and copying stuff).
Copy the configuration/steps from the general openms CI/CD job where it should be working already. Needs to be tested after the permission and jenkins overwrite situation.

**Comment by poshul** [comment]
This has taken the back seat to the FullMultiJob. I hope to get to it after finishing the other.

**Comment by timosachsenberg** [comment]
state of this?

**Comment by timosachsenberg** [comment]
not done - right?

**Comment by jpfeuffer** [comment]
@timosachsenberg Didn't you start on one with me? Before you decided to use Jenkins one more time for the last release?

**Comment by timosachsenberg** [comment]
yes I just mean it is not 100% done.  so would be good to move it to TODO

**Comment by poshul** [comment]
Still having issues with creating the tags, but other than that and updating the website everything is working.

**Comment by poshul** [comment]
Additional dependencies on our existing Jenkins infrastructure showed up when we turned off the UT stuff over the break. I've been working on them.

**Comment by jpfeuffer** [comment]
What's the status? I fixed the website update yesterday. The naming and text of the PR needs to be checked. It says 3.0 but it's a 2.9.1 branch.

**Comment by timosachsenberg** [comment]
do we also need to consider conda?

**Comment by poshul** [comment]
I added the creation of a src tarball as an artifact. Tagging (aside from the genericworkflownodes owned repo) works. The release text was getting mangled I've just committed a fix to base64 encode/decode it when writing to and reading from the output. 
Still missing:
- Changelog Params is not supported right now, I'd like to set it up to be created automatically by the action.
- We aren't uploading sha/md5 hashes right now.

**Comment by poshul** [comment]
oh, and the naming right now is a hack, there weren't any release notes associated with 2.9.1 so I'm just using the name and notes from 3.0.0 (see https://github.com/OpenMS/OpenMS/blob/da1f58334530977a03ac5028602b5b1d6e79c3fb/.github/workflows/openms_ci_matrix_full.yml#L171)

**Comment by jpfeuffer** [comment]
Sounds great! I was about to say that the text for the website does not contain newlines. Maybe that's fixed with your base64 trick?
